__tests__/Goals/components/Create-Goals-test.tsx

```diff
@@ -59,13 +68,56 @@ describe('MainScreen', () => {
     expect(userContainer).toBeNull();
   });
 
-  test.skip('navigates to FormScreen when "Create" button is pressed', () => {
-    const navigateMock = jest.fn();
-    const { getByText } = render(
-      <MembersPage navigation={{ push: navigateMock }} />,
+  test('navigates to Goal Screen when "Create Goal" button is pressed', async () => {
+    const spyAlert = jest.spyOn(Alert, 'alert');
+    axios.get.mockResolvedValue({
+      data: { users: mockUsersData, message: 'Users returned successfully!' },
+    });
+    axios.post.mockResolvedValue(mockGoalData);
+
+    const {
+      getByTestId,
+      getByPlaceholderText,
+      getByText,
+      findByTestId,
+      getAllByText,
+    } = render(<CreateGoals navigation={navigationProp} />);
+
+    const createGoalButton = getByText(/create goal/i);
+    const titleInput = getByPlaceholderText(
+      'Enter title max of 50 characters.',
     );
-    const createButton = getByText('Create');
-    fireEvent.press(createButton);
-    expect(navigateMock).toHaveBeenCalledWith('Form screen');
+    const descriptionInput = getByPlaceholderText('Enter max 200 characters.');
+    const selectUserButton = getByTestId('dropdown');
+
+    fireEvent.press(selectUserButton);
+    const userContainer = await findByTestId('user-container');
+    expect(userContainer).toBeTruthy();
+
+    const userInput = getByPlaceholderText('Search User');
+    await waitFor(() => {
+      fireEvent.changeText(userInput, 'test user');
+    });
+    const userItems = getAllByText(/test user/i);
+    expect(userItems).toBeTruthy();
+
+    await waitFor(() => {
+      fireEvent.press(userItems[0]);
+      fireEvent.changeText(titleInput, 'Test Goal');
+      fireEvent.changeText(descriptionInput, 'Test Description');
+    });
+
+    fireEvent.press(createGoalButton);
+
+    await waitFor(() => {
+      expect(Alert.alert).toHaveBeenCalledWith(
+        'Success',
+        `Task has been created and assigned to test user`,
+        [{ text: 'OK', onPress: expect.any(Function) }],
+      );
+    });
+
+    spyAlert.mock.calls?.[0]?.[2]?.[0].onPress!();
```

Comment:
Nit: can you make a variable with good name and then call the `onPress!()` 

For eg: const spyAlertMockCall = spyAlert.mock.calls?.[0]?.[2]?.[0];

This is for better readability of code.

Reply:
Sure, I'll update.